gn3/db/rdf/wiki.py

```diff
@@ -0,0 +1,164 @@
+    BIND (str(?pubmedId) AS ?pmid) .
+    BIND (str(?createTime) AS ?created) .
+}
+""").substitute(prefix=RDF_PREFIXES, symbol=symbol,)
+    results = query_frame_and_compact(
+        query, WIKI_CONTEXT,
+        sparql_uri
+    )
+    data = results.get("data")
+    if not data:
+        return results
+    return results
+
+
+def get_comment_history(comment_id: int, sparql_uri: str) -> dict:
+    """Get all the historical data for a given id"""
+    query = Template("""
+$prefix
+
+CONSTRUCT {
+    ?uid rdfs:label ?symbolName ;
+         gnt:reason ?reason ;
+         gnt:species ?species ;
+         dct:references ?pmid ;
+         foaf:homepage ?weburl ;
+         rdfs:comment ?comment ;
+         foaf:mbox ?email ;
+         gnt:initial ?usercode ;
+         gnt:belongsToCategory ?category ;
+         gnt:hasVersion ?versionId ;
+         dct:created ?created .
+} WHERE {
+    ?symbolId rdfs:label ?symbolName .
+    ?uid rdf:type gnc:GNWikiEntry ;
+         rdfs:comment ?comment ;
+         gnt:symbol ?symbolId ;
+         dct:created ?createTime ;
+         dct:hasVersion ?version ;
+         dct:identifier $comment_id ;
+         dct:identifier ?id_ .
+    OPTIONAL { ?uid gnt:reason ?reason_ } .
+    OPTIONAL {
+        ?uid gnt:belongsToSpecies ?speciesId .
+        ?speciesId gnt:shortName ?species_ .
+    } .
+    OPTIONAL { ?uid dct:references ?pmid . } .
+    OPTIONAL { ?uid foaf:homepage ?weburl_ . } .
+    OPTIONAL { ?uid gnt:initial ?usercode_ . } .
+    OPTIONAL { ?uid foaf:mbox ?email_ . } .
+    OPTIONAL { ?uid gnt:belongsToCategory ?category_ . } .
+    BIND (str(?version) AS ?versionId) .
+    BIND (str(?createTime) AS ?created) .
+    BIND (COALESCE(?reason_, "") AS ?reason) .
+    BIND (COALESCE(?weburl_, "") AS ?weburl) .
+    BIND (COALESCE(?usercode_, "") AS ?usercode) .
+    BIND (COALESCE(?email_, "") AS ?email) .
+    BIND (COALESCE(?species_, "") AS ?species) .
+    BIND (COALESCE(?category_, "") AS ?category) .
+}
+""").substitute(prefix=RDF_PREFIXES, comment_id=comment_id)
+    results = query_frame_and_compact(
+        query, WIKI_CONTEXT,
+        sparql_uri
+    )
+    data = results.get("data")
+    for result in data:
+        categories = result.get("categories") or []
+        if categories and isinstance(categories, str):
+            result["categories"] = [categories]
+        pmids = result.get("pubmed_ids")
+        if pmids and isinstance(pmids, str):
+            result["pubmed_ids"] = [pmids]
+        elif pmids:
+            result["pubmed_ids"] = [int(pmid) for pmid in pmids]
+        else:
+            result["pubmed_ids"] = []
+        result["version"] = int(result["version"])
+    # We manually sort the array, since somehow "ORDER BY" in the
+    # sparql query is jumbled up during framing and compacting.  This
+    # operation isn't heavy since we don't get many versions per wiki
+    # entry.
+    results["data"] = sorted(data, key=lambda d: d["created"], reverse=True)
```

Comment:
The `createtime` column in our db defaults to `0000-00-000`. I suggest we change the key to `(d['version'], d['created'])` instead

Reply:
I deleted this commit and did a force push.